drivers/backup/portworx/portworx.go

```diff
@@ -221,47 +223,203 @@ func (p *portworx) CreateCloudCredential(ctx context.Context, req *api.CloudCred
+		if cloudCredential.GetName() == cloudCredentialName {
+			return cloudCredential.GetUid(), nil
+		}
+	}
+	return "", fmt.Errorf("cloud credential with name '%s' not found for org '%s'", cloudCredentialName, orgID)
+}
+
+// SetMissingCloudCredentialUID sets the missing cloud-credential UID for cloud-credential-related requests
+func (p *portworx) SetMissingCloudCredentialUID(ctx context.Context, req interface{}) (interface{}, error) {
+	switch r := req.(type) {
+	case *api.CloudCredentialInspectRequest:
+		if r.GetUid() == "" {
+			cloudCredentialUid, err := p.GetCloudCredentialUID(ctx, r.GetOrgId(), r.GetName())
+			if err != nil {
+				return nil, err
+			}
+			r.Uid = cloudCredentialUid
+		}
+		return r, nil
+	case *api.CloudCredentialDeleteRequest:
+		if r.GetUid() == "" {
+			cloudCredentialUid, err := p.GetCloudCredentialUID(ctx, r.GetOrgId(), r.GetName())
+			if err != nil {
+				return nil, err
+			}
+			r.Uid = cloudCredentialUid
+		}
+		return r, nil
+	case *api.CloudCredentialUpdateRequest:
+		if r.GetUid() == "" {
+			cloudCredentialUid, err := p.GetCloudCredentialUID(ctx, r.GetOrgId(), r.GetName())
+			if err != nil {
+				return nil, err
+			}
+			r.Uid = cloudCredentialUid
+		}
+		return r, nil
+	case *api.CloudCredentialOwnershipUpdateRequest:
+		if r.GetUid() == "" {
+			cloudCredentialUid, err := p.GetCloudCredentialUID(ctx, r.GetOrgId(), r.GetName())
+			if err != nil {
+				return nil, err
+			}
+			r.Uid = cloudCredentialUid
+		}
+		return r, nil
+	default:
+		return nil, fmt.Errorf("received unsupported request type %T", req)
+	}
 }
 
 func (p *portworx) CreateCluster(ctx context.Context, req *api.ClusterCreateRequest) (*api.ClusterCreateResponse, error) {
 	return p.clusterManager.Create(ctx, req)
 }
 
 func (p *portworx) UpdateCluster(ctx context.Context, req *api.ClusterUpdateRequest) (*api.ClusterUpdateResponse, error) {
-	return p.clusterManager.Update(ctx, req)
+	reqInterface, err := p.SetMissingClusterUID(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	if reqWithClusterUID, ok := reqInterface.(*api.ClusterUpdateRequest); ok {
+		return p.clusterManager.Update(ctx, reqWithClusterUID)
+	}
+	return nil, fmt.Errorf("expected *api.ClusterUpdateRequest type after filling missing UID, but received %T", reqInterface)
 }
 
 func (p *portworx) InspectCluster(ctx context.Context, req *api.ClusterInspectRequest) (*api.ClusterInspectResponse, error) {
-	return p.clusterManager.Inspect(ctx, req)
+	reqInterface, err := p.SetMissingClusterUID(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	if reqWithClusterUID, ok := reqInterface.(*api.ClusterInspectRequest); ok {
+		return p.clusterManager.Inspect(ctx, reqWithClusterUID)
+	}
+	return nil, fmt.Errorf("expected *api.ClusterInspectRequest type after filling missing UID, but received %T", reqInterface)
 }
 
 func (p *portworx) EnumerateCluster(ctx context.Context, req *api.ClusterEnumerateRequest) (*api.ClusterEnumerateResponse, error) {
```

Comment:
I will add a detailed comment explaining why this had to be done towards the end of the review.